### PR TITLE
Fetch guest credentials and MDS response proto

### PR DESCRIPTION
**Pull request by ChaitanyaKulkarni28**
- Add proto file for MDS response credentials
- Add support for fetching encrypted credentials from MDS and un-marshaling it into MDS response proto.

### CODE REVIEW

File: google_guest_agent/agentcrypto/mtls_mds.go [hunk 1]
```diff
@@ -34,12 +38,19 @@ const (
...
+	// clientCertsKey is the metadata server key at which client identity certificate is exposed.
+	clientCertsKey = "instance/credentials/certs"
 )
 
 var (
 	googleRootCACertUEFIVar = uefi.VariableName{Name: googleRootCACertEFIVarName, GUID: googleGUID}
 )
 
+// MetadataServerClientInterface is the minimum required Metadata Server interface for Guest Agent MTLS bootstrapper.
+type MetadataServerClientInterface interface {
```
Comment (dorileo) [inline google_guest_agent/agentcrypto/mtls_mds.go:50]: Consider moving it to `metadata` package?

File: google_guest_agent/agentcrypto/mtls_mds_test.go [hunk 2]
```diff
@@ -71,3 +73,82 @@ func TestReadAndWriteRootCACertError(t *testing.T) {
...
+
+const (
+	mdsOverride contextKey = iota
+)
+
+// Implements fake metadata server.
+type fakemds struct{}
+
+// GetKey implements fake GetKey MDS method.
+func (s fakemds) GetKey(ctx context.Context, key string) (string, error) {
```
Comment (dorileo) [inline google_guest_agent/agentcrypto/mtls_mds_test.go:87]: a `fake` implementation candidate?

File: google_guest_agent/agentcrypto/mtls_mds.go (hunk 1 above)
Comment (ChaitanyaKulkarni28) [reply google_guest_agent/agentcrypto/mtls_mds.go]: done

File: google_guest_agent/agentcrypto/mtls_mds_test.go [hunk 3]
```diff
@@ -71,3 +73,82 @@ func TestReadAndWriteRootCACertError(t *testing.T) {
...
+	if _, err := getClientCredentials(ctx, client); err != nil {
+		t.Errorf("getClientCredentials(ctx, client) failed unexpectedly with error: %v", err)
+	}
+}
+
+func TestGetClientCredentialsError(t *testing.T) {
+	ctx := context.Background()
+	client := &fakemds{}
+
+	tests := []struct {
```
[the remark this answers is not in the document]
Comment (ChaitanyaKulkarni28) [reply google_guest_agent/agentcrypto/mtls_mds_test.go]: done

File: google_guest_agent/agentcrypto/mtls_mds_test.go (hunk 2 above)
Comment (ChaitanyaKulkarni28) [reply google_guest_agent/agentcrypto/mtls_mds_test.go]: done

**Pull request merged**